Source/WebKit/NetworkProcess/NetworkSession.cpp

```diff
@@ -714,6 +715,48 @@ void NetworkSession::setEmulatedConditions(std::optional<int64_t>&& bytesPerSeco
 
 #endif // ENABLE(INSPECTOR_NETWORK_THROTTLING)
 
+static double connectionTimesMovingAverage(const Deque<Seconds, 25>& connectionTimes)
+{
+    constexpr double alphaSmoothing { 0.75 };
+    // EWMA:
+    // s_0 = x_0
+    // s_t = a * x_{t-1} + (1 - a) * s_{t-1}
+    //
+    // Where x_0 is the average of all recent connection timings, and alpha is 0.75.
+
+    double average = std::accumulate(connectionTimes.begin(), connectionTimes.end(), Seconds { }).seconds() / connectionTimes.size();
+    std::for_each(connectionTimes.begin(), connectionTimes.end(), [&average, alphaSmoothing](auto&& timing) {
```

Comment:
I'm not sure we get much benefit of using std::for_each over a c++11-style for loop.